src/rudymap/jpm/innerptr.rs

```diff
@@ -157,6 +158,16 @@ macro_rules! make_inner_ptr {
             pub fn take(&mut self) -> InnerPtr<K, V> {
                 ::std::mem::replace(self, InnerPtr::default())
             }
+
+            pub fn memory_usage(&self) -> usize {
+                match *self {
+                    $(
+                        InnerPtr::$type(ref target, ref pop) => {
+                            target.memory_usage() + mem::size_of_val(pop)
```

Comment:
`size_of_val(pop)` is not needed because it is included when the parent node calculates the size of itself.